### PR TITLE
LGA 1755 Upload task failure improvements.

### PR DESCRIPTION
**Pull request by Pheonnexx**
## What does this pull request do?

**Add transaction.atomic and useful error handling**
I have added a transaction.atomic so that if the upload of the advisors is not successful it doesn't commit the deletion of the advisor data which it does at the start.

I have attempted to add error handling around this as well but it needs checking.  At the moment on error it updates the task status too failed, and then revisiting the page it correctly says that the last upload failed.

I considered trying to pass back the SQL error but I doubt that this will be of actual use to the user, I wondering if on failure we could just display a checklist of things to check with the data that commonly happen?

## Any other changes that would benefit highlighting?

Intentionally left blank.

## Checklist

- [ ] Provided JIRA ticket number in the title, e.g. "LGA-152: Sample title"

### CODE REVIEW

**Comment by exonian** [comment]
Sorry didn't get to this last week.
Looking good.
Re errors, yes I think you're right that the SQL error may not be of use to the end user, but otoh if they have something they can even pass on to us, that might be helpful. Alternatively, we could start to build up a bit of error message matching behaviour ie, in pseudocode, "if the sql error contains this, display this, else if it contains this, display this, else display this", which gives us a nice entry point to improve it over time as we investigate or hear about issues with it, we can make them easier to solve next time

File: laalaa/apps/advisers/views.py [hunk 1]
```diff
@@ -152,6 +152,20 @@ class UploadSpreadsheetForm(forms.Form):
     xlfile = forms.FileField(label="Spreadsheet")
 
 
+def import_progressive_advisors(xls_file, user):
```
Comment (exonian) [inline laalaa/apps/advisers/views.py:155]: So... I think the task is not to import progressive advisers, but rather to progressively import advisers... 😅

So I think we'd be better suited with `progressively_import_advisers` or even just `import_advisers`.
Also note the spelling of "adviser" here

**Review by exonian — CHANGES_REQUESTED** [review]
Looking really good

File: laalaa/apps/advisers/views.py (hunk 1 above)
Comment (Pheonnexx) [reply laalaa/apps/advisers/views.py]: amended to import_advisers

Comment (Pheonnexx) [reply laalaa/apps/advisers/views.py]: amended method to `import_advisers`

**Review by exonian — APPROVED** [review]

**Pull request merged**